src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.tsx

```diff
@@ -110,9 +115,13 @@ export default class SentryAppDetailsModal extends AsyncComponent<Props, State>
     return toPermissions(this.props.sentryApp.scopes);
   }
 
-  onInstall() {
+  async onInstall() {
     const {onInstall, closeModal, view} = this.props;
-    onInstall();
```

Comment:
Need to make this async such that `componentWillUnmount` happens AFTER installation so the ordering of events is correct.